### PR TITLE
Send back Replacement mimetype when received

### PR DESCRIPTION
**Pull request by Diggers89**
Add optional header to the adaptation outcome message for the replacement mimetype returned by the ncfs api.

### CODE REVIEW

File: Source/Service/NCFS/NcfsOutcome.cs [hunk 1]
```diff
@@ -5,6 +5,8 @@ namespace Service.NCFS
     public class NcfsOutcome
```
Comment (BigPeeg) [inline Source/Service/NCFS/NcfsOutcome.cs:5]: It doesn't seem right to have an outcome plus a decision, within an outcome

**Review by BigPeeg — COMMENTED** [review]
As discussed the TransactionEventProcessor needs breaking up

File: Source/Service/Messaging/IOutcomeSender.cs [hunk 2]
```diff
@@ -1,7 +1,9 @@
-﻿namespace Service.Messaging
+﻿using System.Collections.Generic;
+
+namespace Service.Messaging
 {
     public interface IOutcomeSender
     {
-        void Send(string status, string fileId, string replyTo);
+        void Send(string status, string fileId, string replyTo, Dictionary<string, string> optionalHeaders = null);
```
[the remark this answers is not in the document]
Comment (Diggers89) [reply Source/Service/Messaging/IOutcomeSender.cs]: Done

File: Source/Service/Messaging/OutcomeSender.cs [hunk 3]
```diff
@@ -53,14 +53,22 @@ public void Dispose()
...
         {
             var headers = new Dictionary<string, object>()
                 {
                     { "file-id", fileId },
                     { "file-outcome", status },
                 };
 
+            if (optionalHeaders != null)
+            {
+                foreach (var header in optionalHeaders)
```
[the remark this answers is not in the document]
Comment (Diggers89) [reply Source/Service/Messaging/OutcomeSender.cs:66]: As discussed. Unable to use Concat on `Dictionary<string, object>()`

File: Source/Service/NCFS/NcfsOutcome.cs (hunk 1 above)
Comment (Diggers89) [reply Source/Service/NCFS/NcfsOutcome.cs:5]: Will address this in the refactor

**Pull request merged**